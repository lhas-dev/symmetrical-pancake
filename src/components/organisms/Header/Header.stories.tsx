import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Header } from "./Header";

export default {
  title: "Storybook/Organisms/Header",
  component: Header,
  argTypes: {},
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />;

export const Default = Template.bind({});
Default.args = {};
