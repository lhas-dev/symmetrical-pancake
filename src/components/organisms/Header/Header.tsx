import styled from "styled-components";
import { Icon } from "components/atoms/Icon/Icon";
import {
  Breadcrumb,
  Wrapper as BreadcrumbWrapper,
} from "components/molecules/Breadcrumb/Breadcrumb";
import { Avatar } from "components/molecules/Avatar/Avatar";
import { LogoutButton } from "components/molecules/Header/LogoutButton/LogoutButton";
import { UserInfos } from "components/molecules/Header/UserInfos/UserInfos";
import { NotificationButton } from "components/molecules/Header/NotificationButton/NotificationButton";
import { useDevice } from "hooks/useDevice";
import { Container } from "components/atoms/Container/Container";

interface HeaderProps {}

const Wrapper = styled.header`
  width: 100%;
  min-height: 56px;
  padding-left: 16px;
  padding-right: 16px;
  background: #ffffff;
  border-bottom: 1px solid #ddd;
  display: flex;

  ${Container} {
    align-items: center;
    justify-content: space-between;
  }

  ${BreadcrumbWrapper} {
    margin-left: 20px;
    margin-right: auto;
  }

  @media (min-width: 992px) {
    background: #f5f5f5;
  }
`;

const RightCol = styled.div`
  display: flex;
  align-items: center;
`;

export const Header = (_props: HeaderProps) => {
  const device = useDevice();
  const breadcrumbItems = [
    {
      label: "PÃ¡gina",
    },
  ];

  return (
    <Wrapper>
      <Container>
        <Icon src="DashboardIcon" />
        <Breadcrumb items={breadcrumbItems} />
        <RightCol>
          {device.isDesktop() && <NotificationButton />}
          <Avatar
            src="https://picsum.photos/200"
            name="John Doe"
            hasNotification
          />
          {device.isDesktop() && (
            <>
              <UserInfos name="Silvia Costa" points="4.200" />
              <LogoutButton />
            </>
          )}
        </RightCol>
      </Container>
    </Wrapper>
  );
};
