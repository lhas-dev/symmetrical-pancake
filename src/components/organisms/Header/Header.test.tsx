import { render, act } from "@testing-library/react";
import { Header } from "./Header";

test("renders Header with valid params on mobile devices", async () => {
  Object.defineProperty(window.navigator, "userAgent", {
    value: "Android",
    writable: true,
  });

  const tree = await act(async () => {
    render(<Header />);
  });
  expect(tree).toMatchSnapshot();
});

test("renders Header with valid params on desktop devices", async () => {
  Object.defineProperty(window.navigator, "userAgent", {
    value: "Chrome",
    writable: true,
  });

  const tree = await act(async () => {
    render(<Header />);
  });
  expect(tree).toMatchSnapshot();
});
