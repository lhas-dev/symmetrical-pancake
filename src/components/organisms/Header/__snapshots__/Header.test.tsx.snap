// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header with valid params on desktop devices 1`] = `undefined`;

exports[`renders Header with valid params on mobile devices 1`] = `undefined`;
